File: step17httprequest2/src/main/java/com/example/step17httprequest2/MainActivity.java
package com.example.step17httprequest2;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Map;

public class MainActivity extends AppCompatActivity implements Util.RequestListener{
    EditText editText;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText=findViewById(R.id.editText);

        EditText inputUrl=findViewById(R.id.inputUrl);
        //요청 버튼을 클릭했을때 동작할 준비
        Button requestBtn=findViewById(R.id.requestBtn);
        requestBtn.setOnClickListener(v->{
            //입력한 요청 url 을 읽어온다.
            String url=inputUrl.getText().toString();
            //Util 클래스의 static 메소드를 활용해서 GET 방식 요청하고 결과 받아오기
            Util.sendGetRequest(999, url, null, this);
        });
    }

    @Override
    public void onSuccess(int requestId, Map<String, Object> result) {
        if(requestId == 999){
            // Map 에 data 라는 키값으로 담긴 String type 을 읽어온다.
            String data=(String)result.get("data");
            editText.setText(data);
        }
    }

    @Override
    public void onFail(int requestId, Map<String, Object> result) {

    }
}

File: step17httprequest3/src/main/java/com/example/step17httprequest3/MainActivity.kt
package com.example.step17httprequest3

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import org.json.JSONObject

class MainActivity : AppCompatActivity() , Util.RequestListener{
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //EditText 객체의 참조값 얻어오기
        val editText=findViewById<EditText>(R.id.inputMsg)
        //Button 객체의 참조값 얻어와서 리스너 등록
        val sendBtn=findViewById<Button>(R.id.sendBtn)
        sendBtn.setOnClickListener{
            //EditText 에 입력한 문자열을 읽어와서
            val msg=editText.text.toString()
            Util.sendPostRequest(
                999,
                "http://59.18.221.178:9000/boot07/api/send",
                mapOf("msg" to msg), //"msg" 라는 키값으로 입력한 메세지를 담은 Map
                this
            )

        }
    }

    override fun onSuccess(requestId: Int, result: Map<String, Any?>?) {
        if(requestId == 999){
            //웹서버가 응답한 json 문자열 { }
            val jsonStr = result?.get("data").toString()
            Log.d("#### json 문자열 ####", jsonStr)
            // JSONObject 객체를 생성하면서 생성자의 인자로 json 문자열을 전달한다.
            val obj=JSONObject(jsonStr)
            // key 값을 이용해서 Boolean, String, Int 값을 추출할수 있다.
            val isSuccess=obj.getBoolean("isSuccess")
            val response=obj.getString("response")
            val num=obj.getInt("num")
        }
    }

    override fun onFail(requestId: Int, result: Map<String, Any?>?) {

    }
}